### ISSUE TITLE
Ngsw (cli 1.6) caches everything

### ISSUE THREAD

**Issue by aurnot** [angular/angular-cli#8761]
### Versions
```
Angular CLI: 1.6.0-rc.2
Node: 6.11.5
OS: linux x64
Angular: 5.0.5
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router, service-worker

@angular/cli: 1.6.0-rc.2
@angular-devkit/build-optimizer: 0.0.34
@angular-devkit/core: 0.0.22
@angular-devkit/schematics: 0.0.39
@ngtools/json-schema: 1.1.0
@ngtools/webpack: 1.9.0-rc.2
@schematics/angular: 0.1.9
typescript: 2.4.2
webpack: 3.10.0
```

### Repro steps

1. Create a new app with service worker
``` ng new myapp --service-worker ```

2. Add an external resource to index.html (for example the material icons font)
```html
  <!-- in the head -->
  <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">

  <!-- somewhere in body -->
  <h1>Hello <span style="font-family: 'Material Icons'">cake</span></h1>
```

3. Build the app for prodcution with service worker
``` ng build --prod```

### Observed behavior
Upon first visit, everything is fetched from network as expected
![image](https://user-images.githubusercontent.com/1771805/33681625-66a6aba4-dac5-11e7-90e6-d4faaee949d5.png)
1. The app assets, including the added 4. material icons font
2. The service worker and its config
3. The resources to be cached are fetched again by the service worker

Going offline, and reloading the page, everything is served directly from the service worker, including the webfont that it should not even know about!
![image](https://user-images.githubusercontent.com/1771805/33681992-70f48364-dac6-11e7-8abd-0a36511f57dd.png)
1. Offline mode is enabled
2. The webfont apparently comes from the service worker
3. The webfont indeed works :s

### Desired behavior
Resources not listed in ngsw config should not be cached. This example uses a webfont, but in my actual app API calls are being cached too :(

### Mention any other details that might be useful (optional)
* In the network tab, beneylu.dev is a localhost alias with https enabled
* The ngsw-config.json, the generated ngsw-worker.js and ngsw.json have not been edited

**Comment by alxhub** [comment]
Hello @aurnot,

Unfortunately, this is just DevTools being confusing. All "(from ServiceWorker)" means is that the Service Worker chose to answer that request. It doesn't mean that the Service Worker chose to serve cached content in response.

Due to `@angular/service-worker`'s design, it unconditionally answers all requests while it's active.

If you notice in your devtools screenshot, the two font requests (the original one for the CSS and the one for the woff2 file) appear twice, once as "(from ServiceWorker)", and the second with a gear icon prefixing the file name. This gear icon indicates a _network_ request made by the Service Worker. In this case, `@angular/service-worker` has determined that the font isn't cached as part of the manifest, and has accordingly made its own request to the network.

I believe this still works in Offline mode because of some issues with DevTools and the SW Offline mode - I need to debug this further.

Does that make sense? I'm going to close this as it's not a bug with `@angular/service-worker` or the CLI, but feel free to discuss with me further.

**Comment by aurnot** [comment]
@alxhub Thanks for your answers. Yes this makes sense :) I also noticed the two requests (one from SW and one with the gear icon), but initially thought it was just the SW serving the cached asset, and then checking online if a newer version was available.

When actually going offline (unplugging the ethernet cable), the font requests fail as expected : 
![image](https://user-images.githubusercontent.com/1771805/33705975-af496b4a-db32-11e7-8f28-1f37d35e9543.png)

So this is just an issue with the DevTools Offline mode.
Thanks again 👍

**Comment by oliveti** [comment]
Hi,

I know this ticket is closed but with cli 1.7 I observe similar issue, the service worker is caching everything.

And, it is really caching everything. It means, even if I unplug the cable, I still have a response to the http calls.

I'm using the vanilla ngsw config.
